### PR TITLE
[Filters] Added the ability to hide the clear filter button

### LINKED ISSUE #2989: [Filters] Make the 'Clear' button on a ChoiceList optional
## Feature request summary

Make the clear button on a filter disappear if I need it to!

## Rationale

On a filter where one option must always be selected, I don't want to have the 'Clear' button there especially when it is never clickable. 

Screen Shot 2020-05-14 at 3 24 42 PM

^^ This is a screenshot of my current example for wanting this feature. My filter is a Choicelist with radio buttons where there will  always be a selection made. I think it makes the situation a lil less comfortable for the user to have to see a disabled feature they will never interact with.

---

🌟 Feature requests that are not yet planned will be closed. We then use the issue’s [:+1: upvotes](https://help.github.com/articles/about-conversations-on-github/) to track and set priorities. See the [contribution guidelines](https://github.com/Shopify/polaris-react/blob/master/.github/CONTRIBUTING.md) for more information.

### PR DESCRIPTION
**Pull request by sanaasy**
### WHY are these changes introduced?

Fixes #2989 

### WHAT is this pull request doing?

Adding the ability to remove the clear button. 
Screen Shot 2020-07-23 at 3 33 45 PM

## 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  const [taggedWith, setTaggedWith] = useState(null);
  const [queryValue, setQueryValue] = useState(null);

  const handleTaggedWithChange = useCallback(
    (value) => setTaggedWith(value),
    [],
  );
  const handleQueryValueChange = useCallback(
    (value) => setQueryValue(value),
    [],
  );

  const handleTaggedWithRemove = useCallback(() => setTaggedWith(null), []);
  const handleQueryValueRemove = useCallback(() => setQueryValue(null), []);

  const handleClearAll = useCallback(() => {
    handleTaggedWithRemove();
    handleQueryValueRemove();
  }, [handleQueryValueRemove, handleTaggedWithRemove]);

  const filters = [
    {
      key: 'taggedWith',
      label: 'Tagged with',
      filter: (
        <TextField
          label="Tagged with"
          value={taggedWith}
          onChange={handleTaggedWithChange}
          labelHidden
        />
      ),
      shortcut: true,
      clearButton: false
    },
  ];

  const appliedFilters = !isEmpty(taggedWith)
    ? [
      {
        key: 'taggedWith',
        label: disambiguateLabel('taggedWith', taggedWith),
        onRemove: handleTaggedWithRemove,
      },
    ]
    : [];

  return (
    <div style={{ height: '568px' }}>
      <Card>
        <ResourceList
          resourceName={{ singular: 'customer', plural: 'customers' }}
          filterControl={
            <Filters
              queryValue={queryValue}
              filters={filters}
              appliedFilters={appliedFilters}
              onQueryChange={handleQueryValueChange}
              onQueryClear={handleQueryValueRemove}
              onClearAll={handleClearAll}
            >
              <div style={{ paddingLeft: '8px' }}>
                <Button
                  disabled
                  onClick={() => console.log('New filter saved')}
                >
                  Save
                </Button>
              </div>
            </Filters>
          }
          items={[
            {
              id: 341,
              url: 'customers/341',
              name: 'Mae Jemison',
              location: 'Decatur, USA',
            },
            {
              id: 256,
              url: 'customers/256',
              name: 'Ellen Ochoa',
              location: 'Los Angeles, USA',
            },
          ]}
          renderItem={(item) => {
            const { id, url, name, location } = item;
            const media = <Avatar customer size="medium" name={name} />;

            return (
              <ResourceList.Item
                id={id}
                url={url}
                media={media}
                accessibilityLabel={`View details for ${name}`}
              >
                <h3>
                  <TextStyle variation="strong">{name}</TextStyle>
                </h3>
                <div>{location}</div>
              </ResourceList.Item>
            );
          }}
        />
      </Card>
    </div>
  );

  function disambiguateLabel(key, value) {
    switch (key) {
      case 'taggedWith':
        return `Tagged with ${value}`;
      default:
        return value;
    }
  }

  function isEmpty(value) {
    if (Array.isArray(value)) {
      return value.length === 0;
    } else {
      return value === '' || value == null;
    }
  }
}


```

</details>

### 🎩 checklist

* [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [ ] Updated the component's `README.md` with documentation changes
* [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
* [ ] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the Polaris UI kit

### CODE REVIEW

File: src/components/Filters/Filters.tsx [hunk 1]
```diff
@@ -54,6 +54,10 @@ export interface FilterInterface {
   shortcut?: boolean;
   /** Whether or not the filter is disabled */
   disabled?: boolean;
+  /**
+   * @default true
+   * Whether or not the clear button is displayed */
+  clearButton?: boolean;
```
Comment (BPScott) [inline src/components/Filters/Filters.tsx:60]: optional parameters that default to true behave somewhat weirdly. This is one of those rare cases where we want a negative in a prop name.

`hideClearButton` or something like that.


Alternatively is there a way we can auto hide this in certain circumstances without needing the extra prop? "don't show a clear if there are no selected filters" or something like that

**Comment by alex-page** [comment]
If we change the clear button to be `hideClearButton` I am happy to approve this PR!

File: src/components/Filters/Filters.tsx (hunk 1 above)
Comment (sanaasy) [reply src/components/Filters/Filters.tsx]: (sorry for the delay) @emmaorhun and I fixed up your first comment, swell suggestion, and we tried a lot on the second bit but it was a bit of a hassle and can get done in a future PR as a separate issue!

File: src/components/Filters/Filters.tsx [hunk 2]
```diff
@@ -55,6 +55,10 @@ export interface FilterInterface {
   shortcut?: boolean;
   /** Whether or not the filter is disabled */
   disabled?: boolean;
+  /**
+   * @default true
```
Comment (kyledurand) [inline src/components/Filters/Filters.tsx:59]: Why should this default to true?  It's a breaking change if so and might come be unexpected behaviour for consumers?

File: src/components/Filters/Filters.tsx [hunk 3]
```diff
@@ -540,27 +544,34 @@ class FiltersInner extends Component<CombinedProps, State> {
 
   private generateFilterMarkup(filter: FilterInterface) {
     const i18n = this.props.i18n;
+    const isClearButtonShown =
+      filter.hideClearButton === undefined ? true : !filter.hideClearButton;
```
Comment (kyledurand) [inline src/components/Filters/Filters.tsx:548]: This reads a bit awkward, I think it would be simplified by defaulting to false

**Review by kyledurand — CHANGES_REQUESTED** [review]
Just a couple of comments but I don't think we really need to change the default behaviour here. 

Can you also fix up the formatting issues? Looks like your editor added spaces around `{}` for some reason. Otherwise I think this is looking good and thanks for adding an example 👍

**Comment by alex-page** [comment]
@sanaasy @emmaorhun this is so close, let me know if you all need help to ship this ✨

**Review by alex-page — APPROVED** [review]
@sanaasy I made some small edits, thank you so much for your contribution. Let me know if you want to go through anything ✨

**Review by alex-page — APPROVED** [review]

**Pull request merged**